x-pack/plugins/security/public/account_management/user_profile/user_profile_api_client.ts

```diff
@@ -10,9 +10,34 @@ import { Subject } from 'rxjs';
 
 import type { HttpStart } from '@kbn/core/public';
 
-import type { GetUserProfileResponse, UserProfileData } from '../../../common';
+import type { GetUserProfileResponse, UserProfile, UserProfileData } from '../../../common';
 
-const USER_PROFILE_URL = '/internal/security/user_profile';
+/**
+ * Parameters for the get user profile for the current user API.
+ */
+export interface UserProfileGetCurrentParams {
+  /**
+   * By default, get API returns user information, but does not return any user data. The optional "dataPath"
+   * parameter can be used to return personal data for this user (within `kibana` namespace only).
+   */
+  dataPath: string;
+}
```

Comment:
__note:__ I'm not super happy about having an entire object for the optional `dataPath` parameter on the clien-side, but I decided to keep it consistent with the server-side counterpart where it makes more sense.